verilog/formatting/token_annotator.cc

```diff
@@ -241,6 +242,10 @@ static WithReason<int> SpacesRequiredBetween(
     // TODO(fangism): make this behavior configurable
     if (right.format_token_enum == FormatTokenType::binary_operator &&
         InRangeLikeContext(right_context)) {
+      if (style.compact_non_declaration_ranges && !InDeclaredDimensions(right_context)) {
+        return {0, "Remove space before binary operator inside [] for indexing."};
```

Comment:
Specify that this is targeting indexing/bit-selection uses of [].

Reply:
Done